src/detect-sameip.c

```diff
@@ -143,45 +142,28 @@ static int DetectSameipSigTest01(void)
     p2 = UTHBuildPacketSrcDst(buf, buflen, IPPROTO_TCP, "1.2.3.4", "4.3.2.1");
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
 
     de_ctx->sig_list = SigInit(de_ctx,
                                      "alert tcp any any -> any any "
                                      "(msg:\"Testing sameip\"; sameip; sid:1;)");
-    if (de_ctx->sig_list == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx->sig_list);
 
     SigGroupBuild(de_ctx);
     DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p1);
-    if (PacketAlertCheck(p1, 1) == 0) {
-        printf("sid 2 did not alert, but should have: ");
-        goto cleanup;
-    }
+    FAIL_IF(PacketAlertCheck(p1, 1) == 0);
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p2);
-    if (PacketAlertCheck(p2, 1) != 0) {
-        printf("sid 2 alerted, but should not have: ");
-        goto cleanup;
-    }
-
-    result = 1;
-
-cleanup:
-    SigGroupCleanup(de_ctx);
-    SigCleanSignatures(de_ctx);
```

Comment:
Q: Why did we remove these cleanup functions? Are they covered elsewhere?

Reply:
DetectEngineCtxFree will call those cleanup functions in suricata/src/detect-engine.c

Reply:
I suggested this, following a feedback received by Victor when I was working with one such a task, last year. :P

In the current history: https://github.com/OISF/suricata/pull/6534#discussion_r739618037

Reply:
Awesome! :)